### PR TITLE
[pickers] fix to default to desktop mode when matchMedia is unavailable

### LINKED ISSUE #4985: [pickers] Can't fire change events on DesktopDatePicker using testing-library/react
Current Behavior 

Using fireEvent.change or userEvent.type to simulate changes doesn't trigger the onChange event. The value never changes. But when testing the component manually in the browser, it works fine.

Expected Behavior 

DesktopDatePicker value should change when using fireEvent.change or userEvent.type in tests

### LINKED ISSUE #5655: [testing-library] TimePicker onChange never trigger
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

I'm trying to test this component with testing-library, but I can't trigger onChange event

This issue is related https://github.com/mui/mui-x/issues/4985

### Expected behavior 🤔

the components has an example to use with some framework to do atomatizated tests 

### Steps to reproduce 🕹

Link to live example:

https://codesandbox.io/s/react-testing-library-timepickerexample-f1wly6?file=/src/__tests__/TimePickerExample.test.js

See the tab tests

### Context 🔦

Integration tests

### Your environment 🌎

All the env information are available in live example 

### Order ID 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by LukasTy**
Fixes #5655
Fixes #4985

In test environments (`jsdom` and alike) `window.matchMedia` is not available, hence all the pickers default to rendering mobile component versions, whereas developer (probably) expects the desktop version to be rendered.

The fact that mobile pickers are rendered might confuse developers writing tests and expecting usual input changing events to work correctly. 

**Note:** This change might introduce minor breaking change for developers having tests written expecting the mobile pickers to be rendered in tests.

I'm open to suggestions and/or guidance on how to proceed with this change.
We could leave it for the next major release as well.

### CODE REVIEW

File: test/karma.conf.js [hunk 1]
```diff
@@ -120,7 +120,14 @@ module.exports = function setKarmaConfig(config) {
...
       chromeHeadless: {
         base: 'ChromeHeadless',
-        flags: ['--no-sandbox'],
+        flags: [
+          '--no-sandbox',
+          // running headless chrome in a virtualized environment forces pointer type to default to `NONE`
+          // to mimic "desktop" environment more correctly we force blink to have `pointer: fine` support
+          // this allows correct pickers behavior, where their rendering depends on this condition
+          // https://github.com/microsoft/playwright/issues/7769#issuecomment-1205106311
+          '--blink-settings=primaryPointerType=4',
```
Comment (LukasTy) [inline test/karma.conf.js:129]: @m4theushw You might be interested to see the solution needed to force headless browser to use fine pointer mode.

**Comment by alexfauquette** [comment]
Looks solid 

What about adding an information block in [responsiveness section](https://mui.com/x/react-date-pickers/time-picker/#responsiveness) with a message such as 

> Some test environments (such as jsdom) do not suport media-query. By default, components will be in desktop mode. To modify this behavior you can fake the `window.matchMedia`.

**Comment by LukasTy** [comment]
@alexfauquette What do you think about [such](https://github.com/mui/mui-x/pull/5684/commits/e6da4df949ee598da45a06f1bfac14c4485923a0) documentation implementation?
https://deploy-preview-5684--material-ui-x.netlify.app/x/react-date-pickers/date-picker/#responsiveness

**Review by alexfauquette — APPROVED** [review]
Sounds to be a good fix 👍

**Pull request merged**